lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -1699,6 +1837,238 @@ def _validate_valid_in(self):
+            body['tracking'] = {
+                'requesterName': module.params['entrust_requester_name'],
+                'requesterEmail': module.params['entrust_requester_email'],
+                'requesterPhone': module.params['entrust_requester_phone']
+            }
+            body['eku'] = module.params['entrust_eku']
+
+            try:
+                # self.renew is set when the current certificate is valid, but expiring too soon
+                if(self.renew):
+                    # A renew doesn't accept certType, it renews with existing cert type
+                    del body['certType']
+                    result = self.ecs_client.RenewCertRequest(
+                        trackingId=self.trackingId,
+                        Body=body
+                    )
+                else:
+                    result = self.ecs_client.NewCertRequest(Body=body)
+                self.trackingId = result.get('trackingId')
+            except RestOperationException as e:
+                module.fail_json(msg='Failed to request new certificate from Entrust (ECS) {0}'.format(to_native(e.message)))
+
+            if self.backup:
+                self.backup_file = module.backup_local(self.path)
+            crypto_utils.write_file(module, to_bytes(result.get('endEntityCert')))
+            self.cert = crypto_utils.load_certificate(self.path, backend=self.backend)
+            self.changed = True
+
+    def check(self, module, perms_required=True):
+        """Ensure the resource is in its desired state."""
+
+        parent_check = super(EntrustCertificate, self).check(module, perms_required)
+
+        try:
+            cert_details = self._get_cert_details()
+        except RestOperationException as e:
+            module.fail_json(msg='Failed to get status of existing certificate from Entrust ECS API. Error of {0}.'.format(to_native(e.message)))
+
+        # Always issue a new certificate if the certificate is suspended or revoked
+        status = cert_details.get('status', False)
+        if status == 'EXPIRED' or status == "SUSPENDED" or status == "REVOKED":
+            return False
+
+        # If the requested cert type was specified and it is for a different certificate type than the initial certificate, a new one is needed
+        if module.params['entrust_cert_type'] and cert_details.get('certType') and module.params['entrust_cert_type'] != cert_details.get('certType'):
+            return False
+
+        # If specified, reissue if the certificate expires within x seconds
+        # and the request is for the same cert type
+        reissue_within = module.params.get('entrust_reissue_if_expires_in')
+        seconds_to_expiry = cert_details.get('secondsToExpiry')
+        if seconds_to_expiry and seconds_to_expiry < reissue_within:
+            self.renew = True
+            self.trackingId = cert_details.get('trackingId')
+            return False
+
+        def _validate_ocsp(module):
```

Comment:
To clarify, you mean available for all **providers** or available for all **backends**?

I didn't test with various non-Entrust OCSP responders, which was why I didn't add more general provider support. I'll pull it out - it would make more sense in any modules dedicated just to validation.

Reply:
Ah sorry. Yes, I meant all providers. All backends is more work, since we would need a pyOpenSSL version. But since we want to deprecate pyOpenSSL support anyway, that sounds like wasted effort :)

Actually, it might make sense to have a new module, openssl_ocsp_info or something like that, for querying OCSP responders, since I would like to split network querying code from regular information retrieval. @MarkusTeufelberger @Spredzy any opinions on this?

Reply:
Might want to give it a more general name if there might be a desire for a CRL fall through in the future - e.g. openssl_revocation_info)